packages/starlight/components/Search.astro

```diff
@@ -163,321 +162,327 @@ if (project.trailingSlash === 'never') dataAttributes['data-strip-trailing-slash
-		text-decoration: none;
-	}
+		#starlight__search .pagefind-ui__result-link {
+			position: unset;
+			--pagefind-ui-text: var(--sl-color-white);
+			font-weight: 600;
+		}
 
-	#starlight__search .pagefind-ui__result-nested .pagefind-ui__result-link::before {
-		content: unset;
-	}
+		#starlight__search .pagefind-ui__result-link:hover {
+			text-decoration: none;
+		}
 
-	#starlight__search .pagefind-ui__result-nested::before {
-		content: '';
-		position: absolute;
-		inset-block: 0;
-		inset-inline-start: var(--sl-search-tree-diagram-inline-start);
-		width: var(--sl-search-tree-diagram-size);
-		background: var(--sl-color-gray-4);
-		-webkit-mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' viewBox='0 0 16 1000' preserveAspectRatio='xMinYMin slice'%3E%3Cpath d='M8 0v1000m6-988H8'/%3E%3C/svg%3E")
-			0% 0% / 100% no-repeat;
-		mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' viewBox='0 0 16 1000' preserveAspectRatio='xMinYMin slice'%3E%3Cpath d='M8 0v1000m6-988H8'/%3E%3C/svg%3E")
-			0% 0% / 100% no-repeat;
-	}
-	#starlight__search .pagefind-ui__result-nested:last-of-type::before {
-		-webkit-mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' stroke-linejoin='round' viewBox='0 0 16 16'%3E%3Cpath d='M8 0v12m6 0H8'/%3E%3C/svg%3E");
-		mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' stroke-linejoin='round' viewBox='0 0 16 16'%3E%3Cpath d='M8 0v12m6 0H8'/%3E%3C/svg%3E");
-	}
+		#starlight__search .pagefind-ui__result-nested .pagefind-ui__result-link::before {
+			content: unset;
+		}
 
-	/* Flip page and tree icons around the vertical axis when in an RTL layout. */
-	[dir='rtl'] .pagefind-ui__result-title::before,
-	[dir='rtl'] .pagefind-ui__result-nested::before {
-		transform: matrix(-1, 0, 0, 1, 0, 0);
-	}
+		#starlight__search .pagefind-ui__result-nested::before {
+			content: '';
+			position: absolute;
+			inset-block: 0;
+			inset-inline-start: var(--sl-search-tree-diagram-inline-start);
+			width: var(--sl-search-tree-diagram-size);
+			background: var(--sl-color-gray-4);
+			-webkit-mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' viewBox='0 0 16 1000' preserveAspectRatio='xMinYMin slice'%3E%3Cpath d='M8 0v1000m6-988H8'/%3E%3C/svg%3E")
+				0% 0% / 100% no-repeat;
+			mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='currentColor' stroke-linecap='round' viewBox='0 0 16 1000' preserveAspectRatio='xMinYMin slice'%3E%3Cpath d='M8 0v1000m6-988H8'/%3E%3C/svg%3E")
+				0% 0% / 100% no-repeat;
+		}
+		#starlight__search .pagefind-ui__result-nested:last-child::before {
```

Comment:
Do you remember why this change was needed? I remember we saw those styles being off in the search modal but don’t remember why only `:last-child` worked now.

Reply:
Great catch, this was changed in #2848 to fix an issue when using the [Pagefind metadata feature](https://pagefind.app/docs/metadata/) where the UI was broken. I totally made a mistake when updating the PR and fixing conflicts, good catch.

As is, if you use the feature, you get the old broken behavior:

<img width="608" alt="SCR-20250415-jbhf" src="https://github.com/user-attachments/assets/4266eefb-a30a-4acc-82c1-7141d38589dd" />

I updated the branch to include the fix which now renders properly:

<img width="608" alt="SCR-20250415-jcnt" src="https://github.com/user-attachments/assets/de1ed51b-e6e3-42e3-bfa5-a60236ad7b16" />